### PR TITLE
docs: improve Claude Code agent configuration and BDD workflow

### LINKED ISSUE #1184: Improve Claude Code agent configuration and documentation
## Summary
Improve the Claude Code agent configuration with better documentation, new commands, and skill refinements.

## Changes
- Add `/refocus` command to realign with BDD workflow
- Update orchestration skill to create task checklists
- Expand repo-sherpa with quick reference map
- Update `/plan` skill to reference specs/README.md
- Clean up AGENTS.md with common mistakes table
- Add specs/README.md for BDD guidance

### PR DESCRIPTION
**Pull request by drewdrewthis**
## Summary
- Add `/refocus` command to realign with BDD workflow when lost
- Update orchestration skill to create task checklists using TaskCreate
- Expand repo-sherpa with quick reference map
- Update `/plan` skill to reference specs/README.md
- Add `specs/README.md` with BDD guidance
- Clean up AGENTS.md with common mistakes table
- Add cross-reference to specs/README.md in TESTING_PHILOSOPHY.md

## Test plan
- [ ] Run `/refocus` and verify it provides BDD realignment steps
- [ ] Run `/plan` and verify it reads specs/README.md first
- [ ] Verify AGENTS.md has clear guidance for spec-first development

Closes #1184

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: specs/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- No gaps - if it's not in the feature file, it's not in scope
+
+### Non-Overlapping Test Coverage
+Each test level has a distinct purpose (see `docs/TESTING_PHILOSOPHY.md`):
+
+| Tag | Purpose | What It Tests |
+|-----|---------|---------------|
+| `@e2e` | Happy paths via real examples | Full system, no mocks |
+| `@integration` | Edge cases, error handling | Module boundaries, external services mocked |
+| `@unit` | Pure logic, branches | Single function/class, collaborators mocked |
```
Comment (rogeriochaves) [inline specs/README.md:27]: there was a small thingie here that got lost from the previous BDD file, my claude would add cases for a11y keyboard navigation and other aria stuff, which I think its great to have as part of the spec for a11y, but extremely hard to test, so I had somehwere saying to mark those as non-testable

File: AGENTS.md [hunk 2]
```diff
@@ -55,12 +79,26 @@ specs/               # BDD feature specs
 | Creating shared types for single-use interfaces | Colocate interfaces with their usage; only extract to `types.ts` when shared across multiple files |
 | Using -- on pnpm tasks, pnpm adds the -- automatically | Using e.g. `pnpm test:unit path/to/file` directly |
 
+## Database
+
+| Common Mistake | Correct Behavior |
+|----------------|------------------|
+| Modifying deployed migrations | Never edit migrations that have been deployed - they are immutable history. Create a new migration instead. (New migrations not yet in production can be fixed before merging) |
+| Hardcoding schema names in migrations | Use unqualified table names (e.g., `"Monitor"` not `"langwatch_db"."Monitor"`) - Prisma uses the schema from connection string |
```
Comment (rogeriochaves) [inline AGENTS.md:87]: 👍
